File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        minPrice, maxProf = float("inf"), 0
        for price in prices:
            if price < minPrice: 
                minPrice = price
            elif price - minPrice > maxProf: 
                maxProf = price - minPrice
        return maxProf

File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int profit = 0, low = prices[0];
        for (int price : prices) {
            profit = max(profit, price - low);
            low = min(price, low);
        }
        return profit;
    }
};